p2p/src/network/kad/stream/p2p_network_kad_stream_reducer.rs

```diff
@@ -214,60 +403,135 @@ impl P2pNetworkKadOutgoingStreamState {
+                        let dispatcher = state_context.into_dispatcher();
+                        dispatcher.push(P2pNetworkKademliaStreamAction::WaitOutgoing {
+                            addr: *addr,
+                            peer_id: *peer_id,
+                            stream_id: *stream_id,
+                        });
+                        dispatcher.push(P2pNetworkKademliaAction::UpdateFindNodeRequest {
+                            addr: *addr,
+                            peer_id: *peer_id,
+                            stream_id: *stream_id,
+                            closest_peers,
+                        });
+                        Ok(())
+                    } else {
+                        todo!();
+                    }
                 } else {
-                    *self = P2pNetworkKadOutgoingStreamState::Error(
+                    *state = P2pNetworkKadOutgoingStreamState::Error(
                         P2pNetworkStreamProtobufError::Message("out of bounds".to_owned()),
                     );
                     Ok(())
                 }
             }
             (
                 P2pNetworkKadOutgoingStreamState::PartialReplyReceived { len, data },
-                P2pNetworkKademliaStreamAction::IncomingData { data: new_data, .. },
+                P2pNetworkKademliaStreamAction::IncomingData {
+                    data: new_data,
+                    stream_id,
+                    peer_id,
+                    addr,
+                },
             ) => {
                 let mut data = data.clone();
                 data.extend_from_slice(&new_data.0);
 
                 if *len > data.len() {
-                    *self =
+                    *state =
                         P2pNetworkKadOutgoingStreamState::PartialReplyReceived { len: *len, data };
                     return Ok(());
                 }
 
-                self.handle_incoming_response(*len, &data)
+                state.handle_incoming_response(*len, &data)?;
+                if let P2pNetworkKadOutgoingStreamState::ResponseIsReady {
+                    data: P2pNetworkKademliaRpcReply::FindNode { closer_peers },
+                } = state
+                {
+                    let closest_peers = closer_peers.clone();
+                    let dispatcher = state_context.into_dispatcher();
+                    dispatcher.push(P2pNetworkKademliaStreamAction::WaitOutgoing {
+                        addr: *addr,
+                        peer_id: *peer_id,
+                        stream_id: *stream_id,
+                    });
+                    dispatcher.push(P2pNetworkKademliaAction::UpdateFindNodeRequest {
+                        addr: *addr,
+                        peer_id: *peer_id,
+                        stream_id: *stream_id,
+                        closest_peers,
+                    });
+                    Ok(())
+                } else {
+                    todo!();
```

Comment:
@0xMimir there is a  `todo!()` here. Is it something that can happen or is this a `bug_condition` ?